packages/synthetix-main/test/integration/modules/MarketManagerModule.test.ts

```diff
@@ -74,14 +74,21 @@ describe('MarketManagerModule', function () {
       await systems().Core.connect(user1).mintUsd(accountId, poolId, collateralAddress(), One);
     });
 
+    // TODO: Atm Anvil seems to fail to parse errors when the
+    // entry point (MockMarket) is not the contract that
+    // emits the event (System via MarketManagerModule).
     it('should not work if user has not approved', async () => {
-      await assertRevert(
-        MockMarket().connect(user1).buySynth(One),
-        `MarketDepositNotApproved(${
-          MockMarket().address
-        }, ${await user1.getAddress()}, ${One.toString()}, "0")`,
-        systems().Core
-      );
+      // await assertRevert(
+      //   MockMarket().connect(user1).buySynth(One),
+      //   `MarketDepositNotApproved("${await user1.getAddress()}", "${
+      //     MockMarket().address
+      //   }", ${One.toString()}, 0)`,
+      //   systems().Core
+      // );
+
+      // TODO: Replace this with the commented check above
+      // when fixed. For now we check that it simply reverts.
+      await assertRevert(MockMarket().connect(user1).buySynth(One));
```

Comment:
This is the problem with Anvil that we found: If the entry point is not the contract that emits the event, Anvil fails to understand the revert reason.

Reply:
This is being tracked here: https://www.notion.so/Cannon-Errors-through-proxies-are-not-correctly-displayed-5dd75d2b447d496d86aa1c05ccc62dcb